modules/core/src/test/java/org/apache/ignite/MavenUtils.java

```diff
@@ -0,0 +1,157 @@
+
+        String jarFileName = String.format("%s-%s.jar", artifactId, version);
+
+        String pathToArtifact = getMavenLocalRepositoryPath() + File.separator +
+            groupId.replace(".", File.separator) + File.separator +
+            artifactId.replace(".", File.separator) + File.separator +
+            version + File.separator + jarFileName;
+
+        if (Files.notExists(Paths.get(pathToArtifact)))
+            downloadArtifact(artifact);
+
+        return pathToArtifact;
+    }
+
+    /**
+     * Gets a path to the Maven local repository.
+     *
+     * @return Path to the Maven local repository.
+     * @throws IOException In case of an error.
+     * @throws InterruptedException In case of an error.
+     */
+    public static String getMavenLocalRepositoryPath() throws IOException, InterruptedException {
+        if (locRepPath == null)
+            locRepPath = defineMavenLocalRepositoryPath();
+
+        return locRepPath;
+    }
+
+    /**
+     * Defines a path to the Maven local repository.
+     *
+     * @return Path to the Maven local repository.
+     * @throws IOException In case of an error.
+     * @throws InterruptedException In case of an error.
+     */
+    private static String defineMavenLocalRepositoryPath() throws IOException, InterruptedException {
+        Process p = exec("mvn help:effective-settings");
+
+        String output = CharStreams.toString(new InputStreamReader(p.getInputStream(), Charsets.UTF_8));
+
+        int endTagPos = output.indexOf("</localRepository>");
+
+        assert endTagPos >= 0 : "Couldn't define path to Maven local repository";
+
+        return output.substring(output.lastIndexOf(">", endTagPos) + 1, endTagPos);
+    }
+
+    /**
+     * Downloads and stores in local repository an artifact with given identifier.
+     *
+     * @param artifact Artifact identifier, must match pattern [groupId:artifactId:version].
+     * @throws IOException In case of an error.
+     * @throws InterruptedException In case of an error.
+     */
+    private static void downloadArtifact(String artifact) throws IOException, InterruptedException {
+        X.println("Download an artifact: " + artifact);
+
+        exec("mvn dependency:get -Dartifact=" + artifact);
+
+        X.println("The artifact was downloaded");
+    }
+
+    /**
+     * Executes given command in operation system.
+     *
+     * @param cmd Command to execute.
+     * @return Process of executed command.
+     * @throws IOException In case of an error.
+     * @throws InterruptedException In case of an error.
+     */
+    private static Process exec(String cmd) throws IOException, InterruptedException {
+        ProcessBuilder pb = new ProcessBuilder();
+        pb.redirectErrorStream(true);
+        pb.command(U.isWindows() ?
+            new String[] {"cmd", "/c", cmd} :
+            new String[] {cmd});
+
+        Process p = pb.start();
+
+        assert p.waitFor() == 0 : "Command=" + cmd + " couldn't be executed: "
```

Comment:
1. Agree, fixed.
2. In case of downloading file via maven it may take significant time, therefore I don't want specify timeout. But this class will be used only in tests and it will be managed by "tests execution timeout".

What do you think about 2?